### PR TITLE
create User model

### PR DESCRIPTION
**Pull request by fellainthewagon**

### CODE REVIEW

File: server/src/errors/apiError.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const {
```
Comment (iantiporuk) [inline server/src/errors/apiError.js:1]: usually, these are 3 different error types

File: server/src/middleware/deserializeUser.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    // if "accessT" expired and we have "refreshT", generate NEW accessT if fefreshT is valid
+    const { refreshToken } = req.cookies;
+
+    if (expired && refreshToken) {
+      const data = tokenService.generateNewAccessToken(refreshToken);
+      if (!data) throw ApiError.Unauthorized(NOT_AUTH);
+
+      res.cookie("accessToken", data.accessToken, {
+        httpOnly: true,
+        maxAge: 24 * 60 * 60 * 1000,
```
Comment (iantiporuk) [inline server/src/middleware/deserializeUser.js:26]: magic

File: server/src/middleware/checkExistUser.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const { CONFLICT } = require("http-status-codes");
```
Comment (iantiporuk) [inline server/src/middleware/checkExistUser.js:1]: why do we need this file middleware?

File: server/src/components/storage/database.js [hunk 4]
```diff
@@ -1,30 +1,29 @@
-const DatabaseException = require("../../errors/databaseException");
 const { Resolution } = require("../../db");
```
Comment (iantiporuk) [inline server/src/components/storage/database.js:1]: don't like this file, let's discuss

File: server/config/index.js [hunk 5]
```diff
@@ -8,9 +8,16 @@ const config = {
...
     ttl: parseInt(process.env.TTL, 10) || 30,
   },
+  clientUrl: "http://localhost:5000",
   redis: { host, port },
   resolutionsStorage: process.env.RESOLUTIONS_STORAGE || "memory",
   queueStorage: process.env.QUEUE_STORAGE || "memory",
+  accessTokenTTL: process.env.ACCESS_TOKEN_AGE || "3h",
+  refreshTokenTTL: process.env.REFRESH_TOKEN_AGE || "30d",
+  accessSecret:
+    process.env.JWT_ACCESS_SECRET || "1CncgarOhEmEeux5SYpCC4NtpL8RU9sc",
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply server/config/index.js:18]: yes! I know that

File: server/src/components/auth/authService.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      const tokens = tokenService.generateTokens({ ...userDto });
+
+      return { user: { ...userDto }, ...tokens };
+    } catch (error) {
+      throw new CatchError(error.message);
+    }
+  }
+
+  async logout() {
+    return jwt.sign({}, "fake secret", {
```
[the remark this answers is not in the document]
Comment (fellainthewagon) [reply server/src/components/auth/authService.js:43]: ))))
just an imitation of some logic on this endpoint

**Pull request merged**